frontend/src/lib/components/TaxonomicFilter/taxonomicFilterLogic.ts

```diff
@@ -70,8 +72,17 @@ export const taxonomicFilterLogic = kea<taxonomicFilterLogicType>({
             (taxonomicFilterLogicKey) => taxonomicFilterLogicKey,
         ],
         taxonomicGroups: [
-            (selectors) => [selectors.currentTeamId],
-            (teamId): TaxonomicFilterGroup[] => [
+            (selectors) => [selectors.currentTeamId, selectors.groupTypes],
+            (teamId, groupTypes): TaxonomicFilterGroup[] => [
+                ...(groupTypes.length > 0
```

Comment:
This belongs to a separate selector - putting it here just makes this selector really hard to follow, give it it's own name.

Also, worth checking the feature flag here as well?